scipy/special/_basic.py

```diff
@@ -2528,6 +2528,15 @@ def comb(N, k, *, exact=False, repetition=False):
 
     """
     if repetition:
+        # Special case: C(n, 0) with repetition = 1 for n >= 0
+        # Without this check, comb(0, 0, repetition=True) would compute
+        # comb(-1, 0) which incorrectly returns 0
+        if exact:
+            if int(k) == k and k == 0 and int(N) == N and N >= 0:
+                return 1
```

Comment:
The `int(k) == k` check is redundant

```suggestion
            if k == 0 and int(N) == N and N >= 0:
                return 1
```